File: java/program/Application.java
class Application extends Software {
    private String category, license;
    public Application(String name, String version, String category, String license) {
        super(name, version);
        this.category = category;
        this.license = license;
    }
    public String getName() { return name; }
    public String getCategory() { return category; }

    // setter
    public void setCategory(String category) { this.category = category; }
    public void setLicense(String license) { this.license = license; }
    
}

File: cpp/program/Application.cpp
#pragma once
#include <iostream>
#include <vector>
using namespace std;

#include "Software.cpp"
class Application : public Software {
    private:
        string category;
        string license;
    
    public:
        Application(string n, string v, string cat, string lic)
            : Software(n, v), category(cat), license(lic) {}
    
        // Getters
        string getCategory() const { return category; }
        string getLicense() const { return license; }

    
        // Setters
        void setCategory(string cat) { category = cat; }
        void setLicense(string lic) { license = lic; }
    };